cmd/metrics-prometheus-collector/cmd/cmd.go

```diff
@@ -98,7 +106,9 @@ func init() {
 	RootCmd.Flags().StringVar(&options.metricsPrometheusCollector, "collector-config-filepath", "", "Path to a MetricsPrometheusCollector json or yaml configuration file.")
 	_ = RootCmd.MarkFlagRequired("collector-config-filepath")
 
-	RootCmd.Flags().StringVar(&options.MetricsBindAddress, "http-addr", ":8080", "Bind address of the webservice exposing the metrics and other endpoints.")
+	RootCmd.Flags().StringVar(&options.MetricsBindAddress, "internal-http-addr", "127.0.0.1:8099", "Bind address of the webservice exposing the metrics and other endpoints.")
+	RootCmd.Flags().MarkHidden("internal-http-addr") // this is scraped internally to create a cacheable response
+	RootCmd.Flags().StringVar(&options.externalBindAddress, "http-addr", ":8080", "Bind address to read the cached metrics.")
```

Comment:
I'm a bit confused by this, we are reusing `MetricsBindAddress` to be the internal address and adding an external one?

I'm assuming this is because the internal endpoint does not use any caching and instead we expose the external one which serves only cached content. Correct?